### PR TITLE
DST bugfix

### PR DESCRIPTION
**Pull request by VeselovAlex**

### CODE REVIEW

File: src/test/java/com/axibase/tsd/api/Util.java [hunk 1]
```diff
@@ -104,13 +104,30 @@ public static Long getMillis(String date) throws ParseException {
...
+    /**
+     * @deprecated Implementation affected by DST bug (#3591)
+     */
+    @Deprecated
     public static String transformDateToServerTimeZone(String date, int offsetMinutes) {
         Calendar instance = Calendar.getInstance();
         instance.setTime(parseDate(date));
         instance.add(Calendar.MINUTE, -offsetMinutes);
         return ISOFormat(instance.getTime());
     }
```
Comment (rmakulov) [inline src/test/java/com/axibase/tsd/api/Util.java]: Do we need to store deprecated code?

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java [hunk 2]
```diff
@@ -47,7 +50,9 @@ public static void installParser() throws URISyntaxException, FileNotFoundExcept
                 Assert.fail("Failed to import parser");
         }
         JSONObject jsonObject = new JSONObject(BaseMethod.queryATSDVersion().readEntity(String.class));
-        offsetMinutes = Integer.parseInt(((JSONObject) ((JSONObject) jsonObject.get("date")).get("timeZone")).get("offsetMinutes").toString());
+        JSONObject timeZoneJson = ((JSONObject) ((JSONObject) jsonObject.get("date")).get("timeZone"));
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: Better use `VersionMethod.queryVersion()` method for these purposes.
It already has all models of version response.

```java
Version version = VersionMethod.queryVersion().readEntity(Version.class);
Integer offsetMinutes = version.getDate().getTimeZone().getOffsetMinutes();
```

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java [hunk 3]
```diff
@@ -288,9 +293,29 @@ public void testFileWithCRLFLineBreak(Method method) throws Exception {
...
     }
 
+    /* #3591 */
+    @Test
+    public void testFileWithCRLineBreakAndDST(Method method) throws Exception {
+        System.out.println(new Date());
+        Entity entity = new Entity("e-cr-dst-parser-ms-2");
+        Metric metric = new Metric("m-cr-dst-parser-ms-2");
+
+        File csvPath = resolvePath(RESOURCE_DIR + File.separator + method.getName() + ".csv");
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: Use `String.format` instead multiple concatination

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java [hunk 4]
```diff
@@ -288,9 +293,29 @@ public void testFileWithCRLFLineBreak(Method method) throws Exception {
         String expectedDate = "2015-03-24T06:17:00.000Z";
 
         assertEquals("Incorrect stored value", "533.9", sample.getV().toString());
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: Can you extract `"533.9" to constant?

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java [hunk 5]
```diff
@@ -231,66 +236,64 @@ public void testTimeRangeInMS(Method method) throws Exception {
...
     }
 
-    /* #3011 */
+    /* #3591 */
     @Test
-    public void testFileWithCRLFLineBreak(Method method) throws Exception {
-        Entity entity = new Entity("e-crlf-parser-ms-4");
-        Metric metric = new Metric("m-crlf-parser-ms-4");
+    public void testFileWithCRLineBreakAndDST(Method method) throws Exception {
+        System.out.println(new Date());
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: What is purpose of it  `sout`? Use logger instead sout, if you want to log some information.

File: src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java [hunk 6]
```diff
@@ -231,66 +236,64 @@ public void testTimeRangeInMS(Method method) throws Exception {
...
+        Entity entity = new Entity("e-cr-dst-parser-ms-2");
+        Metric metric = new Metric("m-cr-dst-parser-ms-2");
 
-        File csvPath = resolvePath(RESOURCE_DIR + File.separator + method.getName() + ".csv");
+        File csvPath = resolvePath(Paths.get(RESOURCE_DIR, method.getName() + ".csv").toString());
```
Comment (unrealwork) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: Is it necessary to name the file as the method? What if we will want to change naming convention, we will need to rename files?

Comment (VeselovAlex) [inline src/test/java/com/axibase/tsd/api/method/csv/CSVUploadTest.java]: It's made for compatibility with old code. Possible to rename if nothing be broken

**Comment by unrealwork** [comment]
Add last changes from master

**Pull request merged**